src/Angular.js

```diff
@@ -125,6 +128,39 @@ var VALIDITY_STATE_PROPERTY = 'validity';
 
 var hasOwnProperty = Object.prototype.hasOwnProperty;
 
+var objectMaxDepthInErrorMessage = 5;
+
+/**
+ * @ngdoc function
+ * @name angular.configureErrorHandling
+ * @module ng
+ * @kind function
+ *
+ * @description Use this function to configure error handling.
+ *
+ * @param {Object=} config An object for defining configuration for error handling. The
+ *     following keys are supported:
+ *
+ * * `objectMaxDepth` - used to stringify objects in error messages until reaching the max depth.
+ * if it is not supplied the default max depth which is 5 will be used instead, or if it is invalid
+ * (less than `1` or not a `Number`) no max depth will be used.
+ *
+ */
+function configureErrorHandling(config) {
+  if (isObject(config)) {
+    objectMaxDepthInErrorMessage = isValidObjectMaxDepth(config.objectMaxDepth) ? config.objectMaxDepth : NaN;
+  }
+}
```

Comment:
And i would like to ask why we need to reset max depth in case of invalid argument ?

``` javascript
function configureErrorHandling(config) {
  if (isObject(config)) {
    objectMaxDepthInErrorMessage = isValidObjectMaxDepth(config.objectMaxDepth) ? config.objectMaxDepth : NaN;
  }
}
```

I suggest another constant called DEFAULT_OBJECT_MAX_DEPTH_IN_ERROR_MESSAGE
and use this value in case of invalid objectMaxDepth

``` javascript
function configureErrorHandling(config) {
  if (isObject(config)) {
    objectMaxDepthInErrorMessage = isValidObjectMaxDepth(config.objectMaxDepth) ? config.objectMaxDepth :DEFAULT_OBJECT_MAX_DEPTH_IN_ERROR_MESSAGE;
  }
}
```

And we can supply large maxDepth like 1000 if we want to stringify all the object ?